src/odemis/acq/milling/millmng.py

```diff
@@ -0,0 +1,221 @@
+# -*- coding: utf-8 -*-
+"""
+@author: Patrick Cleeve
+
+Copyright © 2025 Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License version 2 as published by the Free
+Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+
+### Purpose ###
+
+This module contains classes to control the actions related to the milling.
+
+"""
+import logging
+import threading
+import time
+from concurrent.futures._base import (
+    CANCELLED,
+    FINISHED,
+    RUNNING,
+    CancelledError,
+    Future,
+)
+from typing import List
+
+from odemis import model
+from odemis.acq.acqmng import acquire
+from odemis.acq.drift import align_reference_image
+from odemis.acq.milling.patterns import (
+    RectanglePatternParameters,
+)
+from odemis.acq.milling.tasks import MillingTaskSettings
+from odemis.acq.stream import FIBStream
+from odemis.util import executeAsyncTask
+
+
+class TFSMillingTaskManager:
+    """This class manages running milling tasks."""
+
+    def __init__(self, future: Future, tasks: List[MillingTaskSettings], fib_stream: FIBStream):
+        """
+        :param future: the future that will be executing the task
+        :param tasks: The milling tasks to run (in order)
+        """
+
+        self.fibsem = model.getComponent(role="fibsem")
+        self.tasks = tasks
+
+        # for reference image alignment
+        self.fib_stream = fib_stream
+
+        self._future = future
+        if future is not None:
```

Comment:
Can you drop this `if`? It doesn't seem to be ever called with `future==None`, and a quite a bit of code seem to expect `._future` to be a Future...